signatory-sgx/src/seal_signer.rs

```diff
@@ -0,0 +1,135 @@
+use crate::protocol::{Decode, Encode};
+use crate::seal_data::{Label, SealData};
+use serde::{Deserialize, Serialize};
+
+#[cfg(feature = "sgx")]
+use crate::error::Error;
+#[cfg(feature = "sgx")]
+use crate::seal_data::{seal_key, unseal_key};
+#[cfg(feature = "sgx")]
+use aead::{generic_array::GenericArray, Aead, NewAead};
+#[cfg(feature = "sgx")]
+use aes_gcm_siv::Aes128GcmSiv;
+#[cfg(feature = "sgx")]
+use rand::random;
+#[cfg(feature = "sgx")]
+use signatory::ed25519;
+#[cfg(feature = "sgx")]
+use signatory::public_key::PublicKeyed;
+#[cfg(feature = "sgx")]
+use signatory::signature::Signature;
+#[cfg(feature = "sgx")]
+use signatory::signature::{Signer, Verifier};
+#[cfg(feature = "sgx")]
+use signatory_dalek::{Ed25519Signer, Ed25519Verifier};
+
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+pub struct SealedSigner {
+    sealed_seed: Vec<u8>,
+    seal_data: SealData,
+    label: Label,
+}
+
+#[cfg(feature = "sgx")]
+impl SealedSigner {
+    pub fn new() -> Result<Self, Error> {
+        let label: Label = random();
+        let seed = ed25519::Seed::generate();
+        let raw_seed = seed.as_secret_slice();
+
+        let (eget_key, seal_data) = seal_key(label);
+        let aead = get_algo(&eget_key);
+        let nonce = GenericArray::from_slice(&seal_data.nonce);
+        let sealed_seed = aead
```

Comment:
i guess `label` will be public? should it be put as `aad` in the `Payload`: https://docs.rs/aead/0.1.1/aead/struct.Payload.html#structfield.aad ? Instead of passing only `raw_seed`, pass Payload where `raw_seed` is the message and `label` is the additional associated data

Reply:
yes, I can do this even the `label` is private.